### PR TITLE
extdns: remove OCP mentions from main/master configs

### PR DESCRIPTION
**Pull request by alebedev87**
This PR aligns the main branches with the [release 1.1](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.1.yaml).

### CODE REVIEW

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-e2e-azure-ovn-infoblox-operator

**Comment by alebedev87** [comment]
No rehearsal is needed: rebase was needed due to the build cluster change (https://github.com/openshift/release/pull/35749).

**Comment by alebedev87** [comment]
/pj-rehearse

Second rebase let's rehearse to be sure.

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-e2e-aws-ovn-operator

File: ci-operator/config/openshift/external-dns-operator/OWNERS [hunk 1]
```diff
@@ -9,7 +9,6 @@ approvers:
 - candita
 - frobware
 - knobunc
-- lmzuccarelli
```
Comment (candita) [inline ci-operator/config/openshift/external-dns-operator/OWNERS:12]: You could add gcs278, Grant.

File: ci-operator/config/openshift/external-dns/openshift-external-dns-release-1.1.yaml
Comment (candita) [inline ci-operator/config/openshift/external-dns/openshift-external-dns-release-1.1.yaml:1]: Do some of the integration.name need to be changed from 4.12 to 4.13 here too?

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml [hunk 2]
```diff
@@ -53,15 +53,15 @@ presubmits:
     - ^main$
     - ^main-
     cluster: build03
-    context: ci/prow/e2e-aws-operator
+    context: ci/prow/e2e-aws-ovn-operator
```
Comment (candita) [inline ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:56]: I don't think we add ovn to e2e-aws-operator.

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml [hunk 3]
```diff
@@ -118,22 +118,22 @@ presubmits:
...
-    trigger: (?m)^/test( | .* )e2e-aws-operator,?($|\s.*)
+    trigger: (?m)^/test( | .* )e2e-aws-ovn-operator,?($|\s.*)
   - agent: kubernetes
     always_run: true
     branches:
     - ^main$
     - ^main-
     cluster: build05
-    context: ci/prow/e2e-azure-operator
+    context: ci/prow/e2e-azure-ovn-infoblox-operator
```
Comment (candita) [inline ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:128]: Same as above for no ovn on e2e-azure-operator

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml
Comment (candita) [inline ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:1]: Is this generated or did you manually edit it?

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml [hunk 4]
```diff
@@ -63,7 +49,7 @@ tests:
     make test
   container:
     from: src
-- as: e2e-aws-operator
+- as: e2e-aws-ovn-operator
```
Comment (candita) [inline ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml:52]: The *-operator jobs do not need to have ovn added.

**Comment by candita** [comment]
Tide says PR has a merge conflict on this one too.

File: ci-operator/config/openshift/external-dns/openshift-external-dns-release-1.1.yaml
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns/openshift-external-dns-release-1.1.yaml:1]: Not for `1.1` release, it was done for 4.12.

File: ci-operator/config/openshift/external-dns-operator/OWNERS (hunk 1 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/OWNERS:12]: Done.

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml (hunk 4 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml]: Done.

**Comment by alebedev87** [comment]
/pj-rehearse ack

Infoblox failed due to the expired licences.

**Comment by candita** [comment]
@alebedev87 does test `ci/rehearse/openshift/external-dns-operator/main/e2e-azure-infoblox-operator` fail because of infoblox issues?  Can you figure a way to make it succeed?

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml [hunk 5]
```diff
@@ -120,7 +101,7 @@ tests:
         requests:
           cpu: 100m
     workflow: optional-operators-ci-azure
-- as: e2e-infoblox-operator
+- as: e2e-azure-infoblox-operator
```
Comment (candita) [inline ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml:104]: This is currently failing in CI.

File: ci-operator/config/openshift/external-dns/openshift-external-dns-master.yaml [hunk 6]
```diff
@@ -1,34 +1,20 @@
...
-      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12
   to: external-dns
 promotion:
   name: external-dns
   namespace: external-dns-operator
 releases:
   initial:
     integration:
-      name: "4.12"
+      name: "4.13"
```
Comment (candita) [inline ci-operator/config/openshift/external-dns/openshift-external-dns-master.yaml:12]: Shouldn't this name be the version number of external-dns, not openshift?

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml [hunk 7]
```diff
@@ -125,15 +125,15 @@ presubmits:
     - ^main$
     - ^main-
     cluster: build02
-    context: ci/prow/e2e-azure-operator
+    context: ci/prow/e2e-azure-infoblox-operator
```
Comment (candita) [inline ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:128]: Why add infoblox to this one?

File: ci-operator/config/openshift/external-dns/openshift-external-dns-master.yaml (hunk 6 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns/openshift-external-dns-master.yaml:12]: No, this is the version of OCP cluster where the tests will be run.

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml
Comment (candita) [inline ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:1]: I believe this file is generated, so I won't review.

Comment (candita) [reply ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:1]: Except, I wonder why there is a generated file for one folder but not the other?

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml (hunk 5 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml:104]: Yes, all the tests were run but failed due to the expired temporary Infoblox licences. Let me recreate the Infoblox instance.

File: ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml
Comment (alebedev87) [reply ci-operator/jobs/openshift/external-dns-operator/openshift-external-dns-operator-main-presubmits.yaml:1]: Yes, the presubmits are generated from the configs (`ci-operator/config/` directory). Not sure why but it seems that the base image changes don't influence the job definitions. There is no much I can do here as it's generated by `make jobs` command which I have to run to make `ci/prow/generated-config` test pass.

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-e2e-azure-infoblox-operator

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml (hunk 5 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml:104]: [Done](https://github.com/openshift/release/pull/35668#issuecomment-1542932895).

**Comment by alebedev87** [comment]
pj-rehearse

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-ci-index-external-dns-operator-bundle

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-e2e-azure-operator

**Comment by alebedev87** [comment]
/pj-rehearse pull-ci-openshift-external-dns-operator-main-e2e-azure-infoblox-operator

**Pull request merged**